bookShop/src/app/state/books.effects.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { EMPTY } from 'rxjs';
+import { map, mergeMap, catchError } from 'rxjs/operators';
+import { BooksService } from '../core/services/books.service';
+
+@Injectable()
+export class BooksEffects {
+  addBlog$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType('[Book List] Add Book'),
+      mergeMap((res: any) =>
+        this.booksService.addBook(res).pipe(
+          map((book) => ({
+            type: '[Books API] Books Deleted Success',
+            payload: book,
+          })),
+          catchError(() => EMPTY)
```

Comment:
What does `EMPTY` mean? I think you can log an error.

Reply:
Fixed